sync/awsdatabricks.py

```diff
@@ -308,7 +307,7 @@ def monitor_cluster(cluster_id: str, polling_period: int = 30) -> None:
         logger.warning("Unable to monitor cluster due to missing cluster log destination - exiting")
 
 
-def _monitor_cluster(
+def _monitor_cluster(  # noqa: C901
```

Comment:
Obviously not ideal, but this addition barely increased the complexity over the line, and I didn't see any obvious factoring that would *actually* make this logic clearer to understand.